Coding-Tracker.Lawang/CodingController.cs

```diff
@@ -0,0 +1,128 @@
+using System.Configuration;
+using System.Data;
+using System.Globalization;
+using Dapper;
+using Microsoft.Data.Sqlite;
+
+namespace Lawang.Coding_Tracker;
+
+public class CodingController
+{
+    private readonly string _connectionString;
+    public CodingController()
+    {
+        _connectionString = ConfigurationManager.ConnectionStrings["sqlite"].ConnectionString;
+    }
+
+    public List<CodingSession> GetAllData()
+    {
+
+        var codingSessions = new List<CodingSession>();
+        try
+        {
+            using IDbConnection connection = new SqliteConnection(_connectionString);
+            string getAllSQL =
+                @"SELECT * FROM CodingSessions";
+
+            using var dbReader = connection.ExecuteReader(getAllSQL);
```

Comment:
Remember! With Dapper, you do not need all this boilerplate code!
You could have done this to simplify your code a lot:
```return connection.Query<CodingSession>("SELECT * FROM CodingSessions").ToList();```

Dapper handles the tedious work of mapping database rows to objects, so you don't need to manually extract data from a SqlDataReader.